client/game/board/Board.svelte

```diff
@@ -2,29 +2,44 @@
 	export let cards = [];
 
 	import Card from "./Card.svelte";
-	import { createEventDispatcher } from "svelte";
+	import { socket } from "../../connectivity.js";
 	import { areCardsEqual, arrayContainsCard, isValidPlay } from "../shared.js";
 
-	const dispatch = createEventDispatcher();
+	const keyboardMap = ["qwertyu", "asdfghj", "zxcvbnm"].map(line => line.split(""));
+
+	function getKey(i) {
+		if (i >= keyboardMap.length * keyboardMap[0].length) {
+			return "";
+		} else {
+			return keyboardMap[i % 3][parseInt(i / 3)].toUpperCase();
+		}
```

Comment:
So, from what I gather from this file, we have a 3x7 (row x column) array of constant characters. Then, each card on the board gets mapped to one of those characters.

Is there a reason for making this a 2D array, rather than a 1D array?

Reply:
Since the keyboard is arranged in this way:

```
QWERTYU
ASDFGHJ
ZXCVBNM
```

I thought it was a natural way to encode the order. Were you thinking the 1D array should be like `QWERTYUASDFGHJZXCVBNM` or `QAZWSXEDCRFVTGBYHNUJM`?